db/migrate/20200325170358_add_domain_and_relative_path_to_automate.rb

```diff
@@ -0,0 +1,103 @@
+require 'ancestry'
+
+class AddDomainAndRelativePathToAutomate < ActiveRecord::Migration[5.1]
+  class MiqAeNamespace < ActiveRecord::Base
+    has_many :ae_classes, :class_name => "MiqAeClass", :foreign_key => :namespace_id
+    has_ancestry
+  end
+
+  class MiqAeClass < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    belongs_to :ae_namespace, :class_name => "MiqAeNamespace", :foreign_key => :namespace_id
+
+    def relative_path
+      parent_ns = ae_namespace.relative_path
+      parent_ns ? "#{parent_ns}/#{name}" : name
+    end
+
+    def domain_id
+      ae_namespace.domain_id || ae_namespace.id
+    end
```

Comment:
since `domain_id` and `relative_path` are now in this schema, can we drop these 2 methods?

Then again, probably doesn't doesn't hurt anything to leave these in.

Reply:
these are used to set the field.